cirq-core/cirq/study/sweepable.py

```diff
@@ -98,8 +102,13 @@ def to_sweep(
     raise TypeError(f'Unexpected sweep-like value: {sweep_or_resolver_list}')
 
 
-def _resolver_to_sweep(resolver: ParamResolver) -> Sweep:
+def _resolver_to_sweep(resolver: ParamResolver, metadata: Optional[dict]) -> Sweep:
     params = resolver.param_dict
     if not params:
         return UnitSweep
-    return Zip(*[Points(key, [cast(float, value)]) for key, value in params.items()])
+    return Zip(
+        *[
+            Points(key, [cast(float, value)], metadata=metadata.get(key) if metadata else None)
```

Comment:
shouldn't metadata be a dictionary here as well? ... that is instead of attaching the value we attach a map with one entry {unit:unit_value}

Reply:
Not necessarily, in most cases metadata is suppose to be a descriptor like  `DeviceParameter`, see an [example](https://github.com/quantumlib/Cirq/blob/72f0542bc0070a9145b52807694d82261147060a/docs/google/devices.md?plain=1#L93)

Reply:
yes but DeviceParameter is an object that can have some structure. an `str` object doesn't have a strucutre ... I'm just worried that we may need to parse it at some point in the future. however if we are adding something with a bit of structure like a map then we won't run into that.